pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -21,26 +21,54 @@ stdenv.mkDerivation rec {
 
   sourceRoot = ".";
 
-  patches = [ ./bin_to_env.patch ];
-
-  postPatch = ''
-    patchShebangs ./compile.sh
-    for d in scripts src/java_tools src/test src/tools third_party/ijar/test tools; do
+  patchPhase = ''
+    for d in \
+      compile.sh \
+      scripts \
+      src/java_tools \
+      src/test \
+      src/tools \
+      third_party/ijar/test \
+      tools \
+      src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt
+    do
       patchShebangs $d
     done
+    for f in \
+      src/main/java/com/google/devtools/build/lib/analysis/CommandHelper.java \
+      src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java \
+      src/test/java/com/google/devtools/build/lib/shell/CommandTest.java \
+      src/test/java/com/google/devtools/build/lib/shell/InterruptibleTest.java \
+      src/test/java/com/google/devtools/build/lib/shell/LoadTest.java \
+      src/test/java/com/google/devtools/build/lib/skylark/SkylarkRuleImplementationFunctionsTest.java \
+      src/test/java/com/google/devtools/build/lib/standalone/StandaloneSpawnStrategyTest.java
+    do
+      substituteInPlace $f \
+        --replace /bin/bash ${bash}/bin/bash \
+        --replace /bin/cat ${coreutils}/bin/cat \
+        --replace /bin/echo ${coreutils}/bin/echo \
+        --replace /bin/false ${coreutils}/bin/false \
+        --replace /bin/pwd ${coreutils}/bin/pwd \
+        --replace /bin/sleep ${coreutils}/bin/sleep \
+        --replace /bin/true ${coreutils}/bin/true
+    done
   '';
 
   buildInputs = [
     stdenv.cc
     stdenv.cc.cc.lib
-    bash
     jdk
     zip
     unzip
     which
     binutils
   ];
 
+  propagatedBuildInputs = [
```

Comment:
Ah then this probably actually is fine, but please add a comment explaining this.